web-bundle/src/main/resources/com/graphhopper/maps/js/main-template.js

```diff
@@ -901,19 +856,71 @@ function routeLatLng(request, doQuery) {
     var buttonToSelectId = request.getProfile();
     // for legacy requests this might be undefined then we just do not select anything
     if (buttonToSelectId)
-    $("button#" + buttonToSelectId.toLowerCase()).addClass("selectprofile");
+        $("button#" + buttonToSelectId.toLowerCase()).addClass("selectprofile");
 
-    var urlForAPI = request.createURL();
     routeResultsDiv.html('<img src="img/indicator.gif"/> Search Route ...');
-    request.doRequest(urlForAPI, createRouteCallback(request, routeResultsDiv, urlForHistory, doZoom));
+    if (request.cmEditorActive) {
+        doCustomRequest(request, routeResultsDiv);
+    } else {
+        var urlForAPI = request.createURL();
+        request.doRequest(urlForAPI, createRouteCallback(request, routeResultsDiv, urlForHistory, doZoom));
+    }
```

Comment:
Here we finally check whether the custom box was active or not and send either the custom or the standard request. Since this logic is now down here, we can use the custom model box when we drag markers, change the marker order, change the vehicle, add/delete points etc.

Reply:
Maybe we should not run the code block above that starts with `if (!doQuery && History.enabled) {` when the custom model editor is active?